typing/outcometree.mli

```diff
@@ -56,6 +56,8 @@ type out_value =
   | Oval_tuple of out_value list
   | Oval_variant of string * out_value option
 
+type out_type_param = string * (bool * bool * bool)
```

Comment:
It would be good to have a record here rather than a triple of bools.

The variance/injectivity-handling code is already quite boolish, and this PR could be an opportunity to improve that situation.  It would at least be good to avoid exacerbating the problem.

Reply:
Up to now, `outcometree.mli` contains no records.
Is it OK to change that incrementally, as suggested ?

Reply:
Yes! Records are good for our health.

Reply:
Seems that I was wrong, there are records in outcometree. However, in this case it seems better to use Asttypes:
```ocaml
type out_type_param = string * (Asttypes.variance * Asttypes.injectivity)
```